sphinx/writers/latex.py

```diff
@@ -1444,7 +1445,9 @@ def visit_entry(self, node):
             self.needs_linetrimming = 1
         if len(node.traverse(nodes.paragraph)) >= 2:
             self.table.has_oldproblematic = True
-        if isinstance(node.parent.parent, nodes.thead):
+        if isinstance(node.parent.parent, nodes.thead) or \
+           node.parent.parent.parent.stubs[cell.col]:
+            # "stubs" list is an attribute of the tgroup element
             if len(node) == 1 and isinstance(node[0], nodes.paragraph) and node.astext() == '':
                 pass
             else:
```

Comment:
then the code does 
```
                self.body.append('\\sphinxstylethead{\\relax ')
                context = '\\unskip}\\relax ' + context
```
and here I must point out that `\sphinxstylethead`'s default is `\textsf`. However **all** such `\textbf`, `\textsf` etc... from LaTeX do *not* accept multiple paragraph input. It would be easy here to do rather
```
                self.body.append('{\\sphinxstylethead\\relax ')
                context = '\\unskip}\\relax ' + context
```
and the default definition of `\sphinxstylethead` would be `\ttfamily`. Unfortunately this is breaking change for people who customize `\sphinxstylehead` as a macro with one argument. I am making this comment in case the issue arises some day that header cells do not accept multiple paragraphs because of general behaviour of LaTeX's styling macros such as `\textbf` etc...